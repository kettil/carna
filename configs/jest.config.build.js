/* eslint-disable import/no-unresolved, node/no-unpublished-require, node/no-missing-require */
module.exports = Object.assign(require('../../../jest.config.js'), {
  coverageReporters: ['text'],
});
