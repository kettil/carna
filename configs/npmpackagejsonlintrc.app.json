{
  "extends": "@kettil/npm-package-json-lint-config/app"
}
