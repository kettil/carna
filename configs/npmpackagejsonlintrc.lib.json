{
  "extends": "@kettil/npm-package-json-lint-config/lib"
}
