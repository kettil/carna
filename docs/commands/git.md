# The `git` tasks

The tasks will be automatically called by husky during commit.

## The `git commit` task

Checks the files in staged with type, prettier and eslint.

### Exit codes

| Exit code | Description |
| --------- | ----------- |
| 0         | Success     |
| 1         | task failed |

## The `git msg` task

Checks the git commit message for a consistent structure.

### Options

| Options | Description                     |
| ------- | ------------------------------- |
| --edit  | Path to the COMMIT_EDITMSG file |

### Exit codes

| Exit code | Description |
| --------- | ----------- |
| 0         | Success     |
| 1         | task failed |

## The `git push` task

Checks the files with type, prettier and eslint.

### Exit codes

| Exit code | Description |
| --------- | ----------- |
| 0         | Success     |
| 1         | task failed |
