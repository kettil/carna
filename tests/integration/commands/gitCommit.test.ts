import * as fs from 'fs';
import { handler } from '../../../src/lib/commands/git';
import { getArgv, getReadFileWithHooksFiles } from '../../shared/configs';

describe('command git with git "commit" hook', () => {
  test('it should work', async () => {
    const result = await handler(getArgv({ hook: 'commit' }));

    expect(result).toBeUndefined();
  });

  test('it should work with hooks', async () => {
    (fs as any).setMockReadFileFiles(getReadFileWithHooksFiles());

    const result = await handler(getArgv({ hook: 'commit' }));

    expect(result).toBeUndefined();
  }, 15_000);
});
