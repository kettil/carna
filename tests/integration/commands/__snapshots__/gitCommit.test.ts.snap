// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`command git with git "commit" hook it should work with hooks: child-process 1`] = `
Object {
  "args": Array [
    "run",
    "precarna:git",
  ],
  "command": "npm",
  "cwd": "/path/to/project",
}
`;

exports[`command git with git "commit" hook it should work with hooks: child-process 2`] = `
Object {
  "args": Array [
    "diff",
    "--staged",
    "--name-only",
  ],
  "command": "git",
  "cwd": "/path/to/project",
}
`;

exports[`command git with git "commit" hook it should work with hooks: child-process 3`] = `
Object {
  "args": Array [
    "ls-files",
    "--exclude-standard",
    "--modified",
    "--others",
  ],
  "command": "git",
  "cwd": "/path/to/project",
}
`;

exports[`command git with git "commit" hook it should work with hooks: child-process 4`] = `
Object {
  "args": Array [
    "run",
    "precarna:analyse",
  ],
  "command": "npm",
  "cwd": "/path/to/project",
}
`;

exports[`command git with git "commit" hook it should work with hooks: child-process 5`] = `
Object {
  "args": Array [
    "--config",
    "/path/to/project/node_modules/carna/configs/prettierrc.json",
    "--write",
    "/path/to/project/src/index.ts",
    "/path/to/project/src/lib/app.ts",
  ],
  "command": "./node_modules/.bin/prettier",
  "cwd": "/path/to/project",
}
`;

exports[`command git with git "commit" hook it should work with hooks: child-process 6`] = `
Object {
  "args": Array [
    "--color",
    "--max-warnings",
    "0",
    "--config",
    "/path/to/project/node_modules/carna/configs/eslintrc.json",
    "--fix",
    "/path/to/project/src/index.ts",
    "/path/to/project/src/lib/app.ts",
  ],
  "command": "./node_modules/.bin/eslint",
  "cwd": "/path/to/project",
}
`;

exports[`command git with git "commit" hook it should work with hooks: child-process 7`] = `
Object {
  "args": Array [
    "--outDir",
    "/path/to/project/build",
    "--project",
    "/path/to/project/tsconfig.json",
    "--noEmit",
    "--isolatedModules",
    "false",
  ],
  "command": "./node_modules/.bin/tsc",
  "cwd": "/path/to/project",
}
`;

exports[`command git with git "commit" hook it should work with hooks: child-process 8`] = `
Object {
  "args": Array [
    "run",
    "postcarna:analyse",
  ],
  "command": "npm",
  "cwd": "/path/to/project",
}
`;

exports[`command git with git "commit" hook it should work with hooks: child-process 9`] = `
Object {
  "args": Array [
    "add",
    "\\"/path/to/project/src/index.ts\\"",
    "\\"/path/to/project/src/lib/app.ts\\"",
  ],
  "command": "git",
  "cwd": "/path/to/project",
}
`;

exports[`command git with git "commit" hook it should work with hooks: child-process 10`] = `
Object {
  "args": Array [
    "run",
    "precarna:test",
  ],
  "command": "npm",
  "cwd": "/path/to/project",
}
`;

exports[`command git with git "commit" hook it should work with hooks: child-process 11`] = `
Object {
  "args": Array [
    "--config",
    "/path/to/project/jest.config.js",
    "--showConfig",
  ],
  "command": "./node_modules/.bin/jest",
  "cwd": "/path/to/project",
}
`;

exports[`command git with git "commit" hook it should work with hooks: child-process 12`] = `
Object {
  "args": Array [
    "--config",
    "/path/to/project/jest.config.js",
    "--colors",
    "--selectProjects",
    "unit",
    "--maxWorkers",
    "50%",
    "--ci",
    "--silent",
    "--json",
    "--outputFile",
    "/path/to/project/coverage/_unit/jest-final.json",
    "--coverage",
    "--coverageDirectory",
    "/path/to/project/coverage/_unit",
    "--coverageReporters",
    "json",
    "--coverageThreshold",
    "'{\\"global\\":{\\"statements\\":0,\\"branches\\":0,\\"functions\\":0,\\"lines\\":0}}'",
  ],
  "command": "./node_modules/.bin/jest",
  "cwd": "/path/to/project",
}
`;

exports[`command git with git "commit" hook it should work with hooks: child-process 13`] = `
Object {
  "args": Array [
    "--config",
    "/path/to/project/jest.config.js",
    "--colors",
    "--selectProjects",
    "e2e",
    "--maxWorkers",
    "50%",
    "--ci",
    "--silent",
    "--json",
    "--outputFile",
    "/path/to/project/coverage/_e2e/jest-final.json",
    "--coverage",
    "--coverageDirectory",
    "/path/to/project/coverage/_e2e",
    "--coverageReporters",
    "json",
    "--coverageThreshold",
    "'{\\"global\\":{\\"statements\\":0,\\"branches\\":0,\\"functions\\":0,\\"lines\\":0}}'",
  ],
  "command": "./node_modules/.bin/jest",
  "cwd": "/path/to/project",
}
`;

exports[`command git with git "commit" hook it should work with hooks: child-process 14`] = `
Object {
  "args": Array [
    "run",
    "postcarna:test",
  ],
  "command": "npm",
  "cwd": "/path/to/project",
}
`;

exports[`command git with git "commit" hook it should work with hooks: child-process 15`] = `
Object {
  "args": Array [
    "run",
    "precarna:license",
  ],
  "command": "npm",
  "cwd": "/path/to/project",
}
`;

exports[`command git with git "commit" hook it should work with hooks: child-process 16`] = `
Object {
  "args": Array [
    "run",
    "postcarna:license",
  ],
  "command": "npm",
  "cwd": "/path/to/project",
}
`;

exports[`command git with git "commit" hook it should work with hooks: child-process 17`] = `
Object {
  "args": Array [
    "run",
    "precarna:deps",
  ],
  "command": "npm",
  "cwd": "/path/to/project",
}
`;

exports[`command git with git "commit" hook it should work with hooks: child-process 18`] = `
Object {
  "args": Array [
    "run",
    "postcarna:deps",
  ],
  "command": "npm",
  "cwd": "/path/to/project",
}
`;

exports[`command git with git "commit" hook it should work with hooks: child-process 19`] = `
Object {
  "args": Array [
    "run",
    "postcarna:git",
  ],
  "command": "npm",
  "cwd": "/path/to/project",
}
`;

exports[`command git with git "commit" hook it should work with hooks: depcheck 1`] = `
Array [
  "/path/to/project",
  Object {
    "detectors": Array [
      [Function],
      [Function],
    ],
    "ignoreDirs": Array [
      "node_modules",
    ],
    "ignoreMatches": Array [
      "@types/node",
      "typescript",
    ],
    "parsers": Object {
      "**/*.js": [Function],
      "**/*.jsx": [Function],
      "**/*.ts": [Function],
      "**/*.tsx": [Function],
    },
    "specials": Array [
      [Function],
      [Function],
      [Function],
      [Function],
      [Function],
    ],
  },
]
`;

exports[`command git with git "commit" hook it should work with hooks: fs 1`] = `
Object {
  "data": "{\\"/path/to/project/src/index.ts\\":{\\"statementMap\\":{\\"0\\":{\\"start\\":{\\"line\\":22,\\"column\\":36},\\"end\\":{\\"line\\":104,\\"column\\":1}},\\"1\\":{\\"start\\":{\\"line\\":23,\\"column\\":23},\\"end\\":{\\"line\\":23,\\"column\\":59}},\\"2\\":{\\"start\\":{\\"line\\":24,\\"column\\":19},\\"end\\":{\\"line\\":24,\\"column\\":45}},\\"3\\":{\\"start\\":{\\"line\\":27,\\"column\\":2},\\"end\\":{\\"line\\":27,\\"column\\":106}},\\"4\\":{\\"start\\":{\\"line\\":29,\\"column\\":22},\\"end\\":{\\"line\\":29,\\"column\\":71}},\\"5\\":{\\"start\\":{\\"line\\":31,\\"column\\":2},\\"end\\":{\\"line\\":33,\\"column\\":3}},\\"6\\":{\\"start\\":{\\"line\\":32,\\"column\\":4},\\"end\\":{\\"line\\":32,\\"column\\":58}},\\"7\\":{\\"start\\":{\\"line\\":35,\\"column\\":2},\\"end\\":{\\"line\\":37,\\"column\\":3}},\\"8\\":{\\"start\\":{\\"line\\":36,\\"column\\":4},\\"end\\":{\\"line\\":36,\\"column\\":60}},\\"9\\":{\\"start\\":{\\"line\\":40,\\"column\\":2},\\"end\\":{\\"line\\":40,\\"column\\":72}},\\"10\\":{\\"start\\":{\\"line\\":43,\\"column\\":2},\\"end\\":{\\"line\\":46,\\"column\\":4}},\\"11\\":{\\"start\\":{\\"line\\":44,\\"column\\":49},\\"end\\":{\\"line\\":44,\\"column\\":83}},\\"12\\":{\\"start\\":{\\"line\\":48,\\"column\\":2},\\"end\\":{\\"line\\":51,\\"column\\":4}},\\"13\\":{\\"start\\":{\\"line\\":49,\\"column\\":45},\\"end\\":{\\"line\\":49,\\"column\\":75}},\\"14\\":{\\"start\\":{\\"line\\":54,\\"column\\":28},\\"end\\":{\\"line\\":54,\\"column\\":66}},\\"15\\":{\\"start\\":{\\"line\\":56,\\"column\\":2},\\"end\\":{\\"line\\":63,\\"column\\":4}},\\"16\\":{\\"start\\":{\\"line\\":59,\\"column\\":8},\\"end\\":{\\"line\\":59,\\"column\\":82}},\\"17\\":{\\"start\\":{\\"line\\":66,\\"column\\":2},\\"end\\":{\\"line\\":69,\\"column\\":4}},\\"18\\":{\\"start\\":{\\"line\\":71,\\"column\\":2},\\"end\\":{\\"line\\":74,\\"column\\":4}},\\"19\\":{\\"start\\":{\\"line\\":77,\\"column\\":2},\\"end\\":{\\"line\\":87,\\"column\\":4}},\\"20\\":{\\"start\\":{\\"line\\":90,\\"column\\":2},\\"end\\":{\\"line\\":95,\\"column\\":3}},\\"21\\":{\\"start\\":{\\"line\\":91,\\"column\\":4},\\"end\\":{\\"line\\":94,\\"column\\":6}},\\"22\\":{\\"start\\":{\\"line\\":97,\\"column\\":2},\\"end\\":{\\"line\\":103,\\"column\\":3}},\\"23\\":{\\"start\\":{\\"line\\":99,\\"column\\":4},\\"end\\":{\\"line\\":99,\\"column\\":89}},\\"24\\":{\\"start\\":{\\"line\\":102,\\"column\\":4},\\"end\\":{\\"line\\":102,\\"column\\":97}}},\\"fnMap\\":{\\"0\\":{\\"name\\":\\"(anonymous_0)\\",\\"decl\\":{\\"start\\":{\\"line\\":22,\\"column\\":36},\\"end\\":{\\"line\\":22,\\"column\\":37}},\\"loc\\":{\\"start\\":{\\"line\\":22,\\"column\\":59},\\"end\\":{\\"line\\":104,\\"column\\":1}},\\"line\\":22},\\"1\\":{\\"name\\":\\"(anonymous_1)\\",\\"decl\\":{\\"start\\":{\\"line\\":44,\\"column\\":37},\\"end\\":{\\"line\\":44,\\"column\\":38}},\\"loc\\":{\\"start\\":{\\"line\\":44,\\"column\\":49},\\"end\\":{\\"line\\":44,\\"column\\":83}},\\"line\\":44},\\"2\\":{\\"name\\":\\"(anonymous_2)\\",\\"decl\\":{\\"start\\":{\\"line\\":49,\\"column\\":35},\\"end\\":{\\"line\\":49,\\"column\\":36}},\\"loc\\":{\\"start\\":{\\"line\\":49,\\"column\\":45},\\"end\\":{\\"line\\":49,\\"column\\":75}},\\"line\\":49},\\"3\\":{\\"name\\":\\"(anonymous_3)\\",\\"decl\\":{\\"start\\":{\\"line\\":58,\\"column\\":29},\\"end\\":{\\"line\\":58,\\"column\\":30}},\\"loc\\":{\\"start\\":{\\"line\\":59,\\"column\\":8},\\"end\\":{\\"line\\":59,\\"column\\":82}},\\"line\\":59}},\\"branchMap\\":{\\"0\\":{\\"loc\\":{\\"start\\":{\\"line\\":31,\\"column\\":2},\\"end\\":{\\"line\\":33,\\"column\\":3}},\\"type\\":\\"if\\",\\"locations\\":[{\\"start\\":{\\"line\\":31,\\"column\\":2},\\"end\\":{\\"line\\":33,\\"column\\":3}},{\\"start\\":{\\"line\\":31,\\"column\\":2},\\"end\\":{\\"line\\":33,\\"column\\":3}}],\\"line\\":31},\\"1\\":{\\"loc\\":{\\"start\\":{\\"line\\":35,\\"column\\":2},\\"end\\":{\\"line\\":37,\\"column\\":3}},\\"type\\":\\"if\\",\\"locations\\":[{\\"start\\":{\\"line\\":35,\\"column\\":2},\\"end\\":{\\"line\\":37,\\"column\\":3}},{\\"start\\":{\\"line\\":35,\\"column\\":2},\\"end\\":{\\"line\\":37,\\"column\\":3}}],\\"line\\":35},\\"2\\":{\\"loc\\":{\\"start\\":{\\"line\\":90,\\"column\\":2},\\"end\\":{\\"line\\":95,\\"column\\":3}},\\"type\\":\\"if\\",\\"locations\\":[{\\"start\\":{\\"line\\":90,\\"column\\":2},\\"end\\":{\\"line\\":95,\\"column\\":3}},{\\"start\\":{\\"line\\":90,\\"column\\":2},\\"end\\":{\\"line\\":95,\\"column\\":3}}],\\"line\\":90},\\"3\\":{\\"loc\\":{\\"start\\":{\\"line\\":97,\\"column\\":2},\\"end\\":{\\"line\\":103,\\"column\\":3}},\\"type\\":\\"if\\",\\"locations\\":[{\\"start\\":{\\"line\\":97,\\"column\\":2},\\"end\\":{\\"line\\":103,\\"column\\":3}},{\\"start\\":{\\"line\\":97,\\"column\\":2},\\"end\\":{\\"line\\":103,\\"column\\":3}}],\\"line\\":97},\\"4\\":{\\"loc\\":{\\"start\\":{\\"line\\":97,\\"column\\":6},\\"end\\":{\\"line\\":97,\\"column\\":38}},\\"type\\":\\"binary-expr\\",\\"locations\\":[{\\"start\\":{\\"line\\":97,\\"column\\":6},\\"end\\":{\\"line\\":97,\\"column\\":21}},{\\"start\\":{\\"line\\":97,\\"column\\":25},\\"end\\":{\\"line\\":97,\\"column\\":38}}],\\"line\\":97}},\\"s\\":{\\"0\\":4,\\"1\\":18,\\"2\\":18,\\"3\\":18,\\"4\\":18,\\"5\\":18,\\"6\\":0,\\"7\\":18,\\"8\\":6,\\"9\\":18,\\"10\\":18,\\"11\\":186,\\"12\\":18,\\"13\\":72,\\"14\\":18,\\"15\\":18,\\"16\\":436,\\"17\\":18,\\"18\\":18,\\"19\\":18,\\"20\\":18,\\"21\\":18,\\"22\\":18,\\"23\\":9,\\"24\\":9},\\"f\\":{\\"0\\":18,\\"1\\":186,\\"2\\":72,\\"3\\":436},\\"b\\":{\\"0\\":[0,18],\\"1\\":[6,12],\\"2\\":[18,0],\\"3\\":[9,9],\\"4\\":[18,9]},\\"path\\":\\"/path/to/project/src/index.ts\\"},\\"/path/to/project/src/lib/app.ts\\":{\\"statementMap\\":{\\"0\\":{\\"start\\":{\\"line\\":10,\\"column\\":36},\\"end\\":{\\"line\\":16,\\"column\\":1}},\\"1\\":{\\"start\\":{\\"line\\":11,\\"column\\":2},\\"end\\":{\\"line\\":11,\\"column\\":55}},\\"2\\":{\\"start\\":{\\"line\\":13,\\"column\\":2},\\"end\\":{\\"line\\":13,\\"column\\":37}},\\"3\\":{\\"start\\":{\\"line\\":15,\\"column\\":2},\\"end\\":{\\"line\\":15,\\"column\\":56}}},\\"fnMap\\":{\\"0\\":{\\"name\\":\\"(anonymous_0)\\",\\"decl\\":{\\"start\\":{\\"line\\":10,\\"column\\":36},\\"end\\":{\\"line\\":10,\\"column\\":37}},\\"loc\\":{\\"start\\":{\\"line\\":10,\\"column\\":59},\\"end\\":{\\"line\\":16,\\"column\\":1}},\\"line\\":10}},\\"branchMap\\":{},\\"s\\":{\\"0\\":0,\\"1\\":0,\\"2\\":0,\\"3\\":0},\\"f\\":{\\"0\\":0},\\"b\\":{},\\"path\\":\\"/path/to/project/src/lib/app.ts\\"}}
",
  "path": "/path/to/project/coverage/coverage-final.json",
}
`;

exports[`command git with git "commit" hook it should work with hooks: istanbul-reports 1`] = `
Array [
  "lcovonly",
  Object {},
]
`;

exports[`command git with git "commit" hook it should work with hooks: istanbul-reports 2`] = `
Array [
  "json-summary",
  Object {},
]
`;

exports[`command git with git "commit" hook it should work with hooks: istanbul-reports 3`] = `
Array [
  "text-summary",
  Object {},
]
`;

exports[`command git with git "commit" hook it should work: child-process 1`] = `
Object {
  "args": Array [
    "diff",
    "--staged",
    "--name-only",
  ],
  "command": "git",
  "cwd": "/path/to/project",
}
`;

exports[`command git with git "commit" hook it should work: child-process 2`] = `
Object {
  "args": Array [
    "ls-files",
    "--exclude-standard",
    "--modified",
    "--others",
  ],
  "command": "git",
  "cwd": "/path/to/project",
}
`;

exports[`command git with git "commit" hook it should work: child-process 3`] = `
Object {
  "args": Array [
    "--config",
    "/path/to/project/node_modules/carna/configs/prettierrc.json",
    "--write",
    "/path/to/project/src/index.ts",
    "/path/to/project/src/lib/app.ts",
  ],
  "command": "./node_modules/.bin/prettier",
  "cwd": "/path/to/project",
}
`;

exports[`command git with git "commit" hook it should work: child-process 4`] = `
Object {
  "args": Array [
    "--color",
    "--max-warnings",
    "0",
    "--config",
    "/path/to/project/node_modules/carna/configs/eslintrc.json",
    "--fix",
    "/path/to/project/src/index.ts",
    "/path/to/project/src/lib/app.ts",
  ],
  "command": "./node_modules/.bin/eslint",
  "cwd": "/path/to/project",
}
`;

exports[`command git with git "commit" hook it should work: child-process 5`] = `
Object {
  "args": Array [
    "--outDir",
    "/path/to/project/build",
    "--project",
    "/path/to/project/tsconfig.json",
    "--noEmit",
    "--isolatedModules",
    "false",
  ],
  "command": "./node_modules/.bin/tsc",
  "cwd": "/path/to/project",
}
`;

exports[`command git with git "commit" hook it should work: child-process 6`] = `
Object {
  "args": Array [
    "add",
    "\\"/path/to/project/src/index.ts\\"",
    "\\"/path/to/project/src/lib/app.ts\\"",
  ],
  "command": "git",
  "cwd": "/path/to/project",
}
`;

exports[`command git with git "commit" hook it should work: child-process 7`] = `
Object {
  "args": Array [
    "--config",
    "/path/to/project/jest.config.js",
    "--showConfig",
  ],
  "command": "./node_modules/.bin/jest",
  "cwd": "/path/to/project",
}
`;

exports[`command git with git "commit" hook it should work: child-process 8`] = `
Object {
  "args": Array [
    "--config",
    "/path/to/project/jest.config.js",
    "--colors",
    "--selectProjects",
    "unit",
    "--maxWorkers",
    "50%",
    "--ci",
    "--silent",
    "--json",
    "--outputFile",
    "/path/to/project/coverage/_unit/jest-final.json",
    "--coverage",
    "--coverageDirectory",
    "/path/to/project/coverage/_unit",
    "--coverageReporters",
    "json",
    "--coverageThreshold",
    "'{\\"global\\":{\\"statements\\":0,\\"branches\\":0,\\"functions\\":0,\\"lines\\":0}}'",
  ],
  "command": "./node_modules/.bin/jest",
  "cwd": "/path/to/project",
}
`;

exports[`command git with git "commit" hook it should work: child-process 9`] = `
Object {
  "args": Array [
    "--config",
    "/path/to/project/jest.config.js",
    "--colors",
    "--selectProjects",
    "e2e",
    "--maxWorkers",
    "50%",
    "--ci",
    "--silent",
    "--json",
    "--outputFile",
    "/path/to/project/coverage/_e2e/jest-final.json",
    "--coverage",
    "--coverageDirectory",
    "/path/to/project/coverage/_e2e",
    "--coverageReporters",
    "json",
    "--coverageThreshold",
    "'{\\"global\\":{\\"statements\\":0,\\"branches\\":0,\\"functions\\":0,\\"lines\\":0}}'",
  ],
  "command": "./node_modules/.bin/jest",
  "cwd": "/path/to/project",
}
`;

exports[`command git with git "commit" hook it should work: depcheck 1`] = `
Array [
  "/path/to/project",
  Object {
    "detectors": Array [
      [Function],
      [Function],
    ],
    "ignoreDirs": Array [
      "node_modules",
    ],
    "ignoreMatches": Array [
      "@types/node",
      "typescript",
    ],
    "parsers": Object {
      "**/*.js": [Function],
      "**/*.jsx": [Function],
      "**/*.ts": [Function],
      "**/*.tsx": [Function],
    },
    "specials": Array [
      [Function],
      [Function],
      [Function],
      [Function],
      [Function],
    ],
  },
]
`;

exports[`command git with git "commit" hook it should work: fs 1`] = `
Object {
  "data": "{\\"/path/to/project/src/index.ts\\":{\\"statementMap\\":{\\"0\\":{\\"start\\":{\\"line\\":22,\\"column\\":36},\\"end\\":{\\"line\\":104,\\"column\\":1}},\\"1\\":{\\"start\\":{\\"line\\":23,\\"column\\":23},\\"end\\":{\\"line\\":23,\\"column\\":59}},\\"2\\":{\\"start\\":{\\"line\\":24,\\"column\\":19},\\"end\\":{\\"line\\":24,\\"column\\":45}},\\"3\\":{\\"start\\":{\\"line\\":27,\\"column\\":2},\\"end\\":{\\"line\\":27,\\"column\\":106}},\\"4\\":{\\"start\\":{\\"line\\":29,\\"column\\":22},\\"end\\":{\\"line\\":29,\\"column\\":71}},\\"5\\":{\\"start\\":{\\"line\\":31,\\"column\\":2},\\"end\\":{\\"line\\":33,\\"column\\":3}},\\"6\\":{\\"start\\":{\\"line\\":32,\\"column\\":4},\\"end\\":{\\"line\\":32,\\"column\\":58}},\\"7\\":{\\"start\\":{\\"line\\":35,\\"column\\":2},\\"end\\":{\\"line\\":37,\\"column\\":3}},\\"8\\":{\\"start\\":{\\"line\\":36,\\"column\\":4},\\"end\\":{\\"line\\":36,\\"column\\":60}},\\"9\\":{\\"start\\":{\\"line\\":40,\\"column\\":2},\\"end\\":{\\"line\\":40,\\"column\\":72}},\\"10\\":{\\"start\\":{\\"line\\":43,\\"column\\":2},\\"end\\":{\\"line\\":46,\\"column\\":4}},\\"11\\":{\\"start\\":{\\"line\\":44,\\"column\\":49},\\"end\\":{\\"line\\":44,\\"column\\":83}},\\"12\\":{\\"start\\":{\\"line\\":48,\\"column\\":2},\\"end\\":{\\"line\\":51,\\"column\\":4}},\\"13\\":{\\"start\\":{\\"line\\":49,\\"column\\":45},\\"end\\":{\\"line\\":49,\\"column\\":75}},\\"14\\":{\\"start\\":{\\"line\\":54,\\"column\\":28},\\"end\\":{\\"line\\":54,\\"column\\":66}},\\"15\\":{\\"start\\":{\\"line\\":56,\\"column\\":2},\\"end\\":{\\"line\\":63,\\"column\\":4}},\\"16\\":{\\"start\\":{\\"line\\":59,\\"column\\":8},\\"end\\":{\\"line\\":59,\\"column\\":82}},\\"17\\":{\\"start\\":{\\"line\\":66,\\"column\\":2},\\"end\\":{\\"line\\":69,\\"column\\":4}},\\"18\\":{\\"start\\":{\\"line\\":71,\\"column\\":2},\\"end\\":{\\"line\\":74,\\"column\\":4}},\\"19\\":{\\"start\\":{\\"line\\":77,\\"column\\":2},\\"end\\":{\\"line\\":87,\\"column\\":4}},\\"20\\":{\\"start\\":{\\"line\\":90,\\"column\\":2},\\"end\\":{\\"line\\":95,\\"column\\":3}},\\"21\\":{\\"start\\":{\\"line\\":91,\\"column\\":4},\\"end\\":{\\"line\\":94,\\"column\\":6}},\\"22\\":{\\"start\\":{\\"line\\":97,\\"column\\":2},\\"end\\":{\\"line\\":103,\\"column\\":3}},\\"23\\":{\\"start\\":{\\"line\\":99,\\"column\\":4},\\"end\\":{\\"line\\":99,\\"column\\":89}},\\"24\\":{\\"start\\":{\\"line\\":102,\\"column\\":4},\\"end\\":{\\"line\\":102,\\"column\\":97}}},\\"fnMap\\":{\\"0\\":{\\"name\\":\\"(anonymous_0)\\",\\"decl\\":{\\"start\\":{\\"line\\":22,\\"column\\":36},\\"end\\":{\\"line\\":22,\\"column\\":37}},\\"loc\\":{\\"start\\":{\\"line\\":22,\\"column\\":59},\\"end\\":{\\"line\\":104,\\"column\\":1}},\\"line\\":22},\\"1\\":{\\"name\\":\\"(anonymous_1)\\",\\"decl\\":{\\"start\\":{\\"line\\":44,\\"column\\":37},\\"end\\":{\\"line\\":44,\\"column\\":38}},\\"loc\\":{\\"start\\":{\\"line\\":44,\\"column\\":49},\\"end\\":{\\"line\\":44,\\"column\\":83}},\\"line\\":44},\\"2\\":{\\"name\\":\\"(anonymous_2)\\",\\"decl\\":{\\"start\\":{\\"line\\":49,\\"column\\":35},\\"end\\":{\\"line\\":49,\\"column\\":36}},\\"loc\\":{\\"start\\":{\\"line\\":49,\\"column\\":45},\\"end\\":{\\"line\\":49,\\"column\\":75}},\\"line\\":49},\\"3\\":{\\"name\\":\\"(anonymous_3)\\",\\"decl\\":{\\"start\\":{\\"line\\":58,\\"column\\":29},\\"end\\":{\\"line\\":58,\\"column\\":30}},\\"loc\\":{\\"start\\":{\\"line\\":59,\\"column\\":8},\\"end\\":{\\"line\\":59,\\"column\\":82}},\\"line\\":59}},\\"branchMap\\":{\\"0\\":{\\"loc\\":{\\"start\\":{\\"line\\":31,\\"column\\":2},\\"end\\":{\\"line\\":33,\\"column\\":3}},\\"type\\":\\"if\\",\\"locations\\":[{\\"start\\":{\\"line\\":31,\\"column\\":2},\\"end\\":{\\"line\\":33,\\"column\\":3}},{\\"start\\":{\\"line\\":31,\\"column\\":2},\\"end\\":{\\"line\\":33,\\"column\\":3}}],\\"line\\":31},\\"1\\":{\\"loc\\":{\\"start\\":{\\"line\\":35,\\"column\\":2},\\"end\\":{\\"line\\":37,\\"column\\":3}},\\"type\\":\\"if\\",\\"locations\\":[{\\"start\\":{\\"line\\":35,\\"column\\":2},\\"end\\":{\\"line\\":37,\\"column\\":3}},{\\"start\\":{\\"line\\":35,\\"column\\":2},\\"end\\":{\\"line\\":37,\\"column\\":3}}],\\"line\\":35},\\"2\\":{\\"loc\\":{\\"start\\":{\\"line\\":90,\\"column\\":2},\\"end\\":{\\"line\\":95,\\"column\\":3}},\\"type\\":\\"if\\",\\"locations\\":[{\\"start\\":{\\"line\\":90,\\"column\\":2},\\"end\\":{\\"line\\":95,\\"column\\":3}},{\\"start\\":{\\"line\\":90,\\"column\\":2},\\"end\\":{\\"line\\":95,\\"column\\":3}}],\\"line\\":90},\\"3\\":{\\"loc\\":{\\"start\\":{\\"line\\":97,\\"column\\":2},\\"end\\":{\\"line\\":103,\\"column\\":3}},\\"type\\":\\"if\\",\\"locations\\":[{\\"start\\":{\\"line\\":97,\\"column\\":2},\\"end\\":{\\"line\\":103,\\"column\\":3}},{\\"start\\":{\\"line\\":97,\\"column\\":2},\\"end\\":{\\"line\\":103,\\"column\\":3}}],\\"line\\":97},\\"4\\":{\\"loc\\":{\\"start\\":{\\"line\\":97,\\"column\\":6},\\"end\\":{\\"line\\":97,\\"column\\":38}},\\"type\\":\\"binary-expr\\",\\"locations\\":[{\\"start\\":{\\"line\\":97,\\"column\\":6},\\"end\\":{\\"line\\":97,\\"column\\":21}},{\\"start\\":{\\"line\\":97,\\"column\\":25},\\"end\\":{\\"line\\":97,\\"column\\":38}}],\\"line\\":97}},\\"s\\":{\\"0\\":4,\\"1\\":18,\\"2\\":18,\\"3\\":18,\\"4\\":18,\\"5\\":18,\\"6\\":0,\\"7\\":18,\\"8\\":6,\\"9\\":18,\\"10\\":18,\\"11\\":186,\\"12\\":18,\\"13\\":72,\\"14\\":18,\\"15\\":18,\\"16\\":436,\\"17\\":18,\\"18\\":18,\\"19\\":18,\\"20\\":18,\\"21\\":18,\\"22\\":18,\\"23\\":9,\\"24\\":9},\\"f\\":{\\"0\\":18,\\"1\\":186,\\"2\\":72,\\"3\\":436},\\"b\\":{\\"0\\":[0,18],\\"1\\":[6,12],\\"2\\":[18,0],\\"3\\":[9,9],\\"4\\":[18,9]},\\"path\\":\\"/path/to/project/src/index.ts\\"},\\"/path/to/project/src/lib/app.ts\\":{\\"statementMap\\":{\\"0\\":{\\"start\\":{\\"line\\":10,\\"column\\":36},\\"end\\":{\\"line\\":16,\\"column\\":1}},\\"1\\":{\\"start\\":{\\"line\\":11,\\"column\\":2},\\"end\\":{\\"line\\":11,\\"column\\":55}},\\"2\\":{\\"start\\":{\\"line\\":13,\\"column\\":2},\\"end\\":{\\"line\\":13,\\"column\\":37}},\\"3\\":{\\"start\\":{\\"line\\":15,\\"column\\":2},\\"end\\":{\\"line\\":15,\\"column\\":56}}},\\"fnMap\\":{\\"0\\":{\\"name\\":\\"(anonymous_0)\\",\\"decl\\":{\\"start\\":{\\"line\\":10,\\"column\\":36},\\"end\\":{\\"line\\":10,\\"column\\":37}},\\"loc\\":{\\"start\\":{\\"line\\":10,\\"column\\":59},\\"end\\":{\\"line\\":16,\\"column\\":1}},\\"line\\":10}},\\"branchMap\\":{},\\"s\\":{\\"0\\":0,\\"1\\":0,\\"2\\":0,\\"3\\":0},\\"f\\":{\\"0\\":0},\\"b\\":{},\\"path\\":\\"/path/to/project/src/lib/app.ts\\"}}
",
  "path": "/path/to/project/coverage/coverage-final.json",
}
`;

exports[`command git with git "commit" hook it should work: istanbul-reports 1`] = `
Array [
  "lcovonly",
  Object {},
]
`;

exports[`command git with git "commit" hook it should work: istanbul-reports 2`] = `
Array [
  "json-summary",
  Object {},
]
`;

exports[`command git with git "commit" hook it should work: istanbul-reports 3`] = `
Array [
  "text-summary",
  Object {},
]
`;
